timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -842,6 +842,151 @@ void declare_potential(py::module &m) {
+                if (compute_du_dp) {
+                    du_dp.assign(batch_size * P, 9999);
+                }
+                std::vector<__int128> u;
+                if (compute_u) {
+                    u.assign(batch_size, 9999);
+                }
+
+                pot.execute_batch_sparse_host(
+                    coords_size,
+                    N,
+                    params_size,
+                    P,
+                    batch_size,
+                    coords_batch_idxs.data(),
+                    params_batch_idxs.data(),
+                    coords.data(),
+                    params.data(),
+                    boxes.data(),
+                    compute_du_dx ? du_dx.data() : nullptr,
+                    compute_du_dp ? du_dp.data() : nullptr,
+                    compute_u ? u.data() : nullptr);
+
+                auto result = py::make_tuple(py::none(), py::none(), py::none());
+                if (compute_du_dx) {
+                    py::array_t<double, py::array::c_style> py_du_dx({batch_size, N, D});
+                    for (unsigned int i = 0; i < du_dx.size(); i++) {
+                        py_du_dx.mutable_data()[i] = FIXED_TO_FLOAT<double>(du_dx[i]);
+                    }
+                    result[0] = py_du_dx;
+                }
+
+                if (compute_du_dp) {
+                    std::vector<ssize_t> pshape(params.shape(), params.shape() + params.ndim());
+                    pshape[0] = batch_size;
+
+                    py::array_t<double, py::array::c_style> py_du_dp(pshape);
+                    for (unsigned int i = 0; i < batch_size; i++) {
+                        pot.du_dp_fixed_to_float(N, P, &du_dp[0] + (i * P), py_du_dp.mutable_data() + (i * P));
+                    }
+                    result[1] = py_du_dp;
+                }
+
+                if (compute_u) {
+                    py::array_t<double, py::array::c_style> py_u(batch_size);
+
+                    for (unsigned int i = 0; i < py_u.size(); i++) {
+                        py_u.mutable_data()[i] = convert_energy_to_fp(u[i]);
+                    }
+                    result[2] = py_u;
+                }
+
+                return result;
+            },
+            py::arg("coords"),
+            py::arg("params"),
+            py::arg("boxes"),
+            py::arg("coords_batch_idxs"),
+            py::arg("params_batch_idxs"),
+            py::arg("compute_du_dx"),
+            py::arg("compute_du_dp"),
+            py::arg("compute_u"),
+            R"pbdoc(
+        Execute the potential over a batch of coordinates and parameters. Similar to execute_batch, except that instead of evaluating the potential on the dense matrix of pairs of coordinates and parameters, this accepts arrays specifying the indices of the coordinates and parameters to use for each evaluation, allowing evaluation of arbitrary elements of the matrix. The total number of evaluations is len(coords_batch_idxs) [= len(params_batch_idxs)].
```

Comment:
Probably good to mention that the ordering of coords/params can have performance impact

Reply:
Added note in https://github.com/proteneer/timemachine/pull/1308/commits/f8bf85a08c91e413f0d1aa36250add097c368aee